docs/mocks-ut-faq.md

```diff
@@ -0,0 +1,65 @@
+### How are the mock files for unit tests organized?
+- The name of the mock file generated will be the same as the name of the `interface` definition.
+
+- Mock files for `interfaces` defined in the `go-controller/vendor` directories are located in the 
+`go-controller/pkg/testing/mocks` directory. The directory structure in the `go-controller/pkg/testing/mocks/` closely 
+mimic the directory structure of  `go-controller/vendor/`.
+
+	e.g; a) The `Cmd` interface defined in the `go-controller/vendor/k8s.io/utils/exec.go` file has its mock generated 
+	in `go-controller/pkg/testing/mocks/k8s.io/utils/exec/Cmd.go` file
+	
+	e.g; b) The `Link` interface defined in the `go-controller/vendor/github.com/vishvananda/netlink/link.go` file has 
+	its mock generated in `go-controller/pkg/testing/mocks/vishvananda/netlink/Link.go` file
+	
+- Mock files for `interfaces` defined in the non vendor directories of the project are located in the `mocks` directory 
+of the same package as where the interface is defined.
+
+	e.g; a) The `ExecRunner` interface defined in `go-controller/pkg/util/ovs.go` file has the its mock generated in 
+	`go-controller/pkg/util/mocks/ExecRunner.go` file.
+	
+	e.g; b) The `SriovNetLibOps` interface defined in `go-controller/pkg/cni/helper_linux.go` file has its mock 
+	generated in `go-controller/pkg/cni/mocks/SriovNetLibOps.go` file.
+	
+### How are the mocks for interfaces to be consumed by unit tests currently generated?
+
+- The vektra/mockery package from https://github.com/vektra/mockery is leveraged to auto generate mocks for defined interfaces.
+
+- Mocks for interfaces can be generated using vektra/mockery in one of two ways:
```

Comment:
should you now add the option for running the script I just merged :) ?

Reply:
> should you now add the option for running the script I just merged :) ?

good point, let me add a section on how to use the scripts to regenerate mocks